node/read_only_node.go

```diff
@@ -0,0 +1,194 @@
+package node
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+type ReadOnlyNode struct {
```

Comment:
Let's try to use the same `Node`, just git it an extra `readOnly bool` attribute. This file has a lot of code copied from `node.go`.